internal/api/connectors/inventory/xjoin.go

```diff
@@ -0,0 +1,129 @@
+package inventory
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/spf13/viper"
+)
+
+type XjoinConnectorClient interface {
+	GetHostDetailsRequest(
+		ctx context.Context,
+		IDs []string,
+		hostsFilter HostFilter,
+		orderBy HOSTS_ORDER_BY,
+		orderHow ORDER_DIR,
+		limit int,
+		offset int,
+	) (response []*HostDetailsResponse, err error)
+
+	GetSystemProfileDetailsRequest(
+		ctx context.Context,
+		IDs []string,
+		hostsFilter HostFilter,
+		orderBy HOSTS_ORDER_BY,
+		orderHow ORDER_DIR,
+		limit int,
+		offset int,
+	) (response []*SystemProfileDetailsResponse, err error)
```

Comment:
I would remove Request from the name of the method.

I would try to remove the pointer here and just have the slice include instances of `SystemProfile` structs.